tools/rabbitmq-queue-migration.sh

```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/bash
+
+set -ex
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+
+RABBITMQ_SERVICES_TO_RESTART=barbican,blazar,cinder,cloudkitty,designate,heat,ironic,keystone,magnum,manila,neutron,nova,octavia
+RABBITMQ_CONTAINER_NAME=rabbitmq
+
+if [[ ! $KAYOBE_CONFIG_PATH ]]; then
+    echo "${RED}Environment variable \$KAYOBE_CONFIG_PATH is not defined"
+    echo "${RED}Ensure your environment is set up to run kayobe commands"
+    exit 2
+fi
+
+if [[ ! "$1" = "--skip-checks" ]]; then
+    # Fail if clocks are not synced
+    if ! ( kayobe overcloud host command run -l controllers -b --command "timedatectl status | grep 'synchronized: yes'" ); then
+        echo "${RED}Failed precheck: Time not synced on controllers"
+        echo "${RED}Use 'timedatectl status' to check sync state"
+        echo "${RED}Either wait for sync or use 'chronyc makestep'"
+        exit 1
+    fi
+    kayobe overcloud service configuration generate --node-config-dir /tmp/rabbit-migration --kolla-tags none
+    # Fail if HA is set or quorum is not
+    if ! ( grep 'om_enable_queue_manager: true' $KOLLA_CONFIG_PATH/globals.yml && \
+       grep 'om_enable_rabbitmq_quorum_queues: true' $KOLLA_CONFIG_PATH/globals.yml && \
+       grep 'om_enable_rabbitmq_transient_quorum_queue: true' $KOLLA_CONFIG_PATH/globals.yml && \
+       grep 'om_enable_rabbitmq_stream_fanout: true' $KOLLA_CONFIG_PATH/globals.yml ); then
+          echo "${RED}Failed precheck: The following must be enabled: om_enable_queue_manager, om_enable_rabbitmq_quorum_queues, om_enable_rabbitmq_transient_quorum_queue, om_enable_rabbitmq_stream_fanout"
+          exit 1
+    fi
```

Comment:
nit: The spacing here is bit awkward. I think this is slightly more readable, but happy to accept as is
```suggestion
    if ! ( grep 'om_enable_queue_manager: true' $KOLLA_CONFIG_PATH/globals.yml && \
           grep 'om_enable_rabbitmq_quorum_queues: true' $KOLLA_CONFIG_PATH/globals.yml && \
           grep 'om_enable_rabbitmq_transient_quorum_queue: true' $KOLLA_CONFIG_PATH/globals.yml && \
           grep 'om_enable_rabbitmq_stream_fanout: true' $KOLLA_CONFIG_PATH/globals.yml ); then
              echo "${RED}Failed precheck: The following must be enabled: om_enable_queue_manager, om_enable_rabbitmq_quorum_queues, om_enable_rabbitmq_transient_quorum_queue, om_enable_rabbitmq_stream_fanout"
              exit 1
    fi
```